cookiecutter/main.py

```diff
@@ -77,17 +77,17 @@ def cookiecutter(
     )
     import_patch = _patch_import_path_for_repo(repo_dir)
 
-    template_name = os.path.basename(os.path.abspath(repo_dir))
+    template_name = Path(repo_dir).resolve().name
 
     if replay:
         with import_patch:
             if isinstance(replay, bool):
                 context = load(config_dict['replay_dir'], template_name)
             else:
-                path, template_name = os.path.split(os.path.splitext(replay)[0])
-                context = load(path, template_name)
+                replay = Path(replay)
+                context = load(str(path.parent), template_name)
```

Comment:
```suggestion
                context = load(str(replay.parent), template_name)
```
should solve failing test after merge